internal/pkg/provisioner/storageos.go

```diff
@@ -90,16 +91,22 @@ func IsProvisionedPVC(k8s client.Client, pvc corev1.PersistentVolumeClaim, names
 		return false, err
 	}
 
+	return IsProvisionedStorageClass(sc, provisioners...), nil
+}
+
+// IsProvisionedStorageClass returns true if the StorageClass was provided by one of
+// the given provisioners.
+func IsProvisionedStorageClass(sc *storagev1.StorageClass, provisioners ...string) bool {
 	// Check if the StorageClass provisioner matches with any of the provided
 	// provisioners.
 	for _, provisioner := range provisioners {
 		if sc.Provisioner == provisioner {
-			// This is a managed volume.
-			return true, nil
+			// This is a managed storage class.
```

Comment:
Seems a little incorrect. Volumes are managed by storageos if the provisioner value is storageos. But storageclasses are not managed by storageos. SC uses storageos provisioner.